src/tools/sssctl/sssctl_logs.c

```diff
@@ -387,6 +384,10 @@ errno_t sssctl_debug_level(struct sss_cmdline *cmdline,
 
 fini:
     talloc_free(ctx);
+    /* pc_config_file is allocated by popt using malloc().
+     * debug_as_string is not allocated but points to the command line. */
+    free((void *)(uintptr_t) pc_config_file);
```

Comment:
You can use `discard_const` macro.

Reply:
Perhaps it should not be declared `const`?

Reply:
> You can use `discard_const` macro.

Sure.

Reply:
> in the call chain there are functions that require a `const char *` and the compiler would complain if passed a 'char *'.

Why?

Reply:
Sorry, it was not `const char *` but `const char **`. By bad.
In that case, you get some error like this:
```
warning: passing argument 3 of ‘get_targets’ from incompatible pointer type [-Wincompatible-pointer-types]
```
and casting it will not solve the problem.

Reply:
> The comment still stands, please use `discard_const` if needed.

Sorry if I missed some use of this macro. Can you please point me to them? I'm really can' t find them. Even using a textual search I only find old usages.